src/S2EExecutionStateRegisters.cpp

```diff
@@ -169,30 +169,40 @@ bool S2EExecutionStateRegisters::readSymbolicRegion(unsigned offset, void *_buf,
 
     // XXX: deal with alignment and overlaps?
 
-    ref<Expr> value = wos->read(offset, size * 8);
-    uint64_t concreteValue;
-    if (!isa<ConstantExpr>(value)) {
-        if (!concretize) {
-            return false;
+    // m_concretizer->concretize require value size <= sizeof(uint64_t), to
+    // support data size > sizeof(uint64_t), we do concretization every
+    // sizeof(uint64_t) bytes in a loop, or we can rewrite m_concretizer->concretize
+    // to remove the sizeof(uint64_t) bytes limitation
+    for (unsigned i = 0; i < size; i += sizeof(uint64_t)) {
+        unsigned csize = (size - i) > sizeof(uint64_t) ? sizeof(uint64_t) : (size - i);
+        ref<Expr> value = wos->read(offset + i, csize * 8);
+        uint64_t concreteValue;
+        if (!isa<ConstantExpr>(value)) {
+            if (!concretize) {
+                return false;
+            }
+
+            size_t regIndex = offset / sizeof(target_ulong);
+            std::string regName = regIndex < (sizeof(regNames) / sizeof(regNames[0]))
+                                      ? regNames[regIndex]
+                                      : "CPUOffset-" + std::to_string(offset);
+            std::string reason = std::string("access to ") + regName + " register from libcpu helper";
```

Comment:
You can just do “access to “ + regName

I don’t think the explicit std::string constructor is necessary.

Reply:
Thanks for your review fixed.